pkg/util/optional/optional.go

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package optional has optional types and functions used by Agent.
+package optional
+
+// Option represents an optional type.
+// By default, no value is set and a call to Get() returns (T{}, false)
+type Option[T any] struct {
+	value T
```

Comment:
Can move the file instead of creating new ? We are losing the git history becoz of the change.

Reply:
I did `git mv` on this file so I'm not sure why it's not showing up here, but I believe the git history is still there on this file:
- [History for optional.go](https://github.com/DataDog/datadog-agent/commits/9ba08ab52bb1ee4b47f718121f98abdb28025ec3/pkg/util/optional/optional.go)
- [History for optional_test.go](https://github.com/DataDog/datadog-agent/commits/9ba08ab52bb1ee4b47f718121f98abdb28025ec3/pkg/util/optional/optional_test.go) (which appears the same as optional.go and I also did `git mv` on)